mrc-ide/hint
# mrc-2795 create structured generic logging

This PR logs structured data which gets shipped to ELK in Json format. 

My solution uses spring-boot logging implementation, which was previously disabled. I believe disabling spring-logging was because slf4j-simple was implemented within postgreSQL which was conflicting with spring logging framework. I upgraded Postgres to the latest version which has now dropped slf4j-simple logging as part of it's dependencies.

Then,  I enabled spring-logging as hint default logging so as to configure logging behaviour.  Config allows  for formatting of logs in Json. The config file `logback-spring.xml`  renders  logs to console as `STDOUT` when in development environment and as Json when not in dev environment.

The intermittent failure has resurfaced again, once we restart the build kite server, everything should be fine. While I have not found a real solution to the stale buildkite state, I think manual reboot fixes the issue for now.


To test normal format:
*  Run the app
*  go to project history page
* edit project note
* then check the logs display normally in app console 


To test Json format
* Change Spring profile in grade.properties 
* follow the above test
* then check logs display in Json

Please speak to Rob to gain access to ELK to see display

A simple visualisation dashboard showing the test log

<img width="1424" alt="Screenshot 2022-05-19 at 14 59 07" src="https://user-images.githubusercontent.com/4240748/169311492-59bf63c9-c99b-4ac1-8969-81f0de3153dd.png">


link to the dashboard, you need to be on side network.
https://logs.dide.ic.ac.uk/app/dashboards#/view/68f9cd60-679d-11eb-9ee9-77ae80d7fb74?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2022-05-19T09:32:08.152Z',to:'2022-05-28T10:02:25.502Z'))

For a better visualisation, we need to create a mock up for visualisation, deciding what to see on hint dashboard.

## Type of version change
_Delete as appropriate_

Major - Minor - Patches - None

## Checklist
